src/background.ts

```diff
@@ -91,6 +93,22 @@ class ShortExt {
     launch() {
         this.interceptRequests();
     }
+
+    // Function to handle extension clicks
+    // Required: browser_action clause in manifest.json to tell the browser that we will need it's events.
+    redirectToBaseURI = (base_uri: String) => {
+        // Event Listener
+        chrome.browserAction.onClicked.addListener(function(tab) 
+            { 
+                // Check if the current tab has some url/event should not be triggered on empty tabs
+                if(tab.hasOwnProperty("url")){
+                    url = tab.url;
+                    chrome
+                    .tabs
+                    .create({ url: `${base_uri}/?long_link=${url}` }, () => {});
```

Comment:
```suggestion
                    .create({ url: `${homepageURL}/?long_link=${currentPageURL}` }, () => {});
```

Reply:
Do we need `() => {}` for the second parameter?

Reply:
It's an optional callback for create method, might be useful if there is a need to get that tab's instance.
Will remove for now.